activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -460,7 +461,12 @@ def failsafe(method, returning: nil)
           yield
         rescue ::Redis::BaseError => e
           handle_exception exception: e, method: method, returning: returning
-          returning
+
+          if @raise_errors
```

Comment:
Still call `handle_exception` to allow for global handlers as well as local ones(rescue).